x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/table/table.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiInMemoryTable, EuiSearchBarProps } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { CoreStart } from 'kibana/public';
+import moment from 'moment';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import useDebounce from 'react-use/lib/useDebounce';
+import useInterval from 'react-use/lib/useInterval';
+import { TableText } from '../';
+import { SessionsMgmtConfigSchema } from '../..';
+import { ActionComplete, UISession } from '../../../../../common/search/sessions_mgmt';
+import { SearchSessionsMgmtAPI } from '../../lib/api';
+import { getColumns } from '../../lib/get_columns';
+import { getAppFilter } from './app_filter';
+import { getStatusFilter } from './status_filter';
+
+const TABLE_ID = 'searchSessionsMgmtTable';
+
+interface Props {
+  core: CoreStart;
+  api: SearchSessionsMgmtAPI;
+  initialTable: UISession[] | null;
+  timezone: string;
+  config: SessionsMgmtConfigSchema;
+}
+
+export function SearchSessionsMgmtTable({
+  core,
+  api,
+  timezone,
+  initialTable,
+  config,
+  ...props
+}: Props) {
+  const [tableData, setTableData] = useState<UISession[]>(initialTable ? initialTable : []);
+  const [isLoading, setIsLoading] = useState(false);
+  const [debouncedIsLoading, setDebouncedIsLoading] = useState(false);
+  const [pagination, setPagination] = useState({ pageIndex: 0 });
+  const [refreshInterval, setRefreshInterval] = useState<number | undefined>(undefined);
+  const showLatestResultsHandler = useRef<Function>();
+
+  // Debounce rendering the state of the Refresh button
+  useDebounce(
+    () => {
+      setDebouncedIsLoading(isLoading);
+    },
+    250,
+    [isLoading]
+  );
+
+  // Convert the config interval to ms
+  useEffect(() => {
+    setRefreshInterval(moment.duration(config.refreshInterval).asMilliseconds());
+  }, [config.refreshInterval]);
+
+  // refresh behavior
+  const doRefresh = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const renderResults = (results: UISession[]) => {
+        setTableData(results);
+      };
+      showLatestResultsHandler.current = renderResults;
+      const results = await api.fetchTableData();
+      if (results) {
+        // Ignore results from any but the latest api call
+        if (showLatestResultsHandler.current === renderResults) renderResults(results);
```

Comment:
Agreed on first part.

What do you mean by?

> I think we also need this for setIsLoading(false) below

Reply:
there is `setIsLoading(false)` at the bottom of this effect and I think there is the same race condition as with `renderResults` but we don't handle it.
it could be that you've fired two requests, when the first completes it set `isLoading` to `false` even though there is 2nd request loading.